src/lib/discovery.ts

```diff
@@ -96,3 +106,185 @@ export const generateClinicalDataFile = async (
 		throw new Error('Failed to retrieve clinical file data');
 	}
 };
+
+type DataDiscoveryESDocument = {
+	donor_id: string;
+	gender: string;
+	submitter_donor_id: string;
+	study_id: string;
+	updated_at: string;
+	program_id: string;
+	vital_status: string;
+	cause_of_death: string;
+	survival_time: number;
+	primary_site: string[];
+	height: number;
+	weight: number;
+	bmi: number;
+	genetic_disorders: string[];
+
+	analyses: {
+		analysis_id: string;
+		analysis_type: string;
+		analysis_version: number;
+		files: {
+			index_file?: { object_id: string };
+			file_id: string;
+			file_type: string;
+			md5sum: string;
+			name: string;
+			object_id: string;
+			size: number;
+		}[];
+		repositories: {
+			code: string;
+			name: string;
+			organization: string;
+			country: string;
+			url: string;
+		}[];
+	}[];
+
+	specimens: { samples: { sample_id: string }[] }[];
+};
+
+type FileRecord = {
+	donor_id: string;
+	program_id: string;
+	analysis_id: string;
+	repositories: string[];
+	sample_ids: string[];
+	file: {
+		object_id: string;
+		type: string;
+		name: string;
+		size: number;
+		md5sum: string;
+		index_file?: string;
+	};
+};
+
+/**
+ * Convert donor centric records into file records
+ * Consumes ES document stream and outputs stream for input to TSV writer stream
+ *
+ * @param stream ES document stream
+ */
+async function* donorsToFiles(
+	stream: AsyncGenerator<DataDiscoveryESDocument[], void, unknown>,
+): AsyncGenerator<FileRecord[], void, unknown> {
```

Comment:
Thanks for clear output typing. Very important, especially on a generator function.